perun-core/src/main/java/cz/metacentrum/perun/core/blImpl/MembersManagerBlImpl.java

```diff
@@ -2328,31 +2322,44 @@ public Member createSponsoredMember(PerunSession session, Vo vo, String namespac
 		}
 		User sponsoredUser = getPerunBl().getUsersManagerBl().createUser(session, user);
 
+		return setSponsoredMember(session, vo, sponsoredUser, namespace, password, sponsor, asyncValidation);
+	}
+
+	@Override
+	public Member setSponsoredMember(PerunSession session, Vo vo, User userToBeSponsored, String namespace, String password, User sponsor, boolean asyncValidation) throws AlreadyMemberException, ExtendMembershipException, UserNotInRoleException, PasswordStrengthException, WrongAttributeValueException, WrongReferenceAttributeValueException, LoginNotExistsException, PasswordCreationFailedException, InvalidLoginException, ExtSourceNotExistsException {
+		//check that sponsoring user has role SPONSOR for the VO
+		if (!getPerunBl().getVosManagerBl().isUserInRoleForVo(session, sponsor, Role.SPONSOR, vo, true)) {
+			throw new UserNotInRoleException("user " + sponsor.getId() + " is not in role SPONSOR for VO " + vo.getId());
+		}
+		String loginAttributeName = PasswordManagerModule.LOGIN_PREFIX + namespace;
+
 		//create the user account in external system
 		Map<String, String> p = new HashMap<>();
```

Comment:
I have just realized that there has to be made one change. We don't want to generate the account in external systems if the user already has it (which might happen now). Here you need to check if the user has `loginAttributeName` attribute with not-null and not-empty value. If not, you will generate the account.